File: specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/examples-java/Fleets_CreateOrUpdate_MinimumSet.java

import com.azure.core.management.SubResource;
import com.azure.resourcemanager.computefleet.models.ApiEntityReference;
import com.azure.resourcemanager.computefleet.models.BaseVirtualMachineProfile;
import com.azure.resourcemanager.computefleet.models.CachingTypes;
import com.azure.resourcemanager.computefleet.models.ComputeProfile;
import com.azure.resourcemanager.computefleet.models.DiskCreateOptionTypes;
import com.azure.resourcemanager.computefleet.models.EvictionPolicy;
import com.azure.resourcemanager.computefleet.models.FleetProperties;
import com.azure.resourcemanager.computefleet.models.ImageReference;
import com.azure.resourcemanager.computefleet.models.LinuxConfiguration;
import com.azure.resourcemanager.computefleet.models.NetworkApiVersion;
import com.azure.resourcemanager.computefleet.models.OperatingSystemTypes;
import com.azure.resourcemanager.computefleet.models.RegularPriorityAllocationStrategy;
import com.azure.resourcemanager.computefleet.models.RegularPriorityProfile;
import com.azure.resourcemanager.computefleet.models.SpotAllocationStrategy;
import com.azure.resourcemanager.computefleet.models.SpotPriorityProfile;
import com.azure.resourcemanager.computefleet.models.StorageAccountTypes;
import com.azure.resourcemanager.computefleet.models.VirtualMachineScaleSetIPConfiguration;
import com.azure.resourcemanager.computefleet.models.VirtualMachineScaleSetIPConfigurationProperties;
import com.azure.resourcemanager.computefleet.models.VirtualMachineScaleSetManagedDiskParameters;
import com.azure.resourcemanager.computefleet.models.VirtualMachineScaleSetNetworkConfiguration;
import com.azure.resourcemanager.computefleet.models.VirtualMachineScaleSetNetworkConfigurationProperties;
import com.azure.resourcemanager.computefleet.models.VirtualMachineScaleSetNetworkProfile;
import com.azure.resourcemanager.computefleet.models.VirtualMachineScaleSetOSDisk;
import com.azure.resourcemanager.computefleet.models.VirtualMachineScaleSetOSProfile;
import com.azure.resourcemanager.computefleet.models.VirtualMachineScaleSetStorageProfile;
import com.azure.resourcemanager.computefleet.models.VmSizeProfile;
import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

/**
 * Samples for Fleets CreateOrUpdate.
 */
public final class Main {
    /*
     * x-ms-original-file: 2024-11-01/Fleets_CreateOrUpdate_MinimumSet.json
     */
    /**
     * Sample code: Fleets_CreateOrUpdate_MinimumSet.
     * 
     * @param manager Entry point to ComputeFleetManager.
     */
    public static void
        fleetsCreateOrUpdateMinimumSet(com.azure.resourcemanager.computefleet.ComputeFleetManager manager) {
        manager.fleets().define("testFleet").withRegion("eastus2euap").withExistingResourceGroup("rgazurefleet")
            .withTags(mapOf("key", "fakeTokenPlaceholder"))
            .withProperties(new FleetProperties()
                .withSpotPriorityProfile(new SpotPriorityProfile().withCapacity(2).withMinCapacity(1)
                    .withEvictionPolicy(EvictionPolicy.DELETE)
                    .withAllocationStrategy(SpotAllocationStrategy.PRICE_CAPACITY_OPTIMIZED).withMaintain(true))
                .withRegularPriorityProfile(new RegularPriorityProfile().withCapacity(2).withMinCapacity(1)
                    .withAllocationStrategy(RegularPriorityAllocationStrategy.LOWEST_PRICE))
                .withVmSizesProfile(Arrays.asList(new VmSizeProfile().withName("Standard_D2s_v3"),
                    new VmSizeProfile().withName("Standard_D4s_v3"), new VmSizeProfile().withName("Standard_E2s_v3")))
                .withComputeProfile(new ComputeProfile()
                    .withBaseVirtualMachineProfile(new BaseVirtualMachineProfile()
                        .withOsProfile(new VirtualMachineScaleSetOSProfile().withComputerNamePrefix("prefix")
                            .withAdminUsername("azureuser").withAdminPassword("fakeTokenPlaceholder")
                            .withLinuxConfiguration(new LinuxConfiguration().withDisablePasswordAuthentication(false)))
                        .withStorageProfile(new VirtualMachineScaleSetStorageProfile()
                            .withImageReference(new ImageReference()
                                .withPublisher("canonical").withOffer("0001-com-ubuntu-server-focal")
                                .withSku("20_04-lts-gen2").withVersion("latest"))
                            .withOsDisk(new VirtualMachineScaleSetOSDisk().withCaching(CachingTypes.READ_WRITE)
                                .withCreateOption(DiskCreateOptionTypes.FROM_IMAGE)
                                .withOsType(OperatingSystemTypes.LINUX)
                                .withManagedDisk(new VirtualMachineScaleSetManagedDiskParameters()
                                    .withStorageAccountType(StorageAccountTypes.STANDARD_LRS))))
                        .withNetworkProfile(new VirtualMachineScaleSetNetworkProfile()
                            .withNetworkInterfaceConfigurations(
                                Arrays.asList(new VirtualMachineScaleSetNetworkConfiguration().withName("vmNameTest")
                                    .withProperties(new VirtualMachineScaleSetNetworkConfigurationProperties()
                                        .withPrimary(true).withEnableAcceleratedNetworking(false)
                                        .withIpConfigurations(Arrays.asList(new VirtualMachineScaleSetIPConfiguration()
                                            .withName("vmNameTest")
                                            .withProperties(new VirtualMachineScaleSetIPConfigurationProperties()
                                                .withSubnet(new ApiEntityReference().withId(
                                                    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworks/{virtualNetworkName}/subnets/{subnetName}"))
                                                .withPrimary(true).withLoadBalancerBackendAddressPools(
                                                    Arrays.asList(new SubResource().withId(
                                                        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/loadBalancers/{loadBalancerName}/backendAddressPools/{backendAddressPoolName}"))))))
                                        .withEnableIPForwarding(true))))
                            .withNetworkApiVersion(NetworkApiVersion.fromString("2022-07-01"))))
                    .withComputeApiVersion("2023-09-01").withPlatformFaultDomainCount(1)))
            .create();
    }

    // Use "Map.of" if available
    @SuppressWarnings("unchecked")
    private static <T> Map<String, T> mapOf(Object... inputs) {
        Map<String, T> map = new HashMap<>();
        for (int i = 0; i < inputs.length; i += 2) {
            String key = (String) inputs[i];
            T value = (T) inputs[i + 1];
            map.put(key, value);
        }
        return map;
    }
}

File: specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/examples-js/Fleets_CreateOrUpdate_MinimumSet.js
const { AzureFleetClient } = require("@azure/arm-computefleet");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to create a Fleet
 *
 * @summary create a Fleet
 * x-ms-original-file: 2024-11-01/Fleets_CreateOrUpdate_MinimumSet.json
 */
async function fleetsCreateOrUpdateMinimumSet() {
  const credential = new DefaultAzureCredential();
  const subscriptionId = "1DC2F28C-A625-4B0E-9748-9885A3C9E9EB";
  const client = new AzureFleetClient(credential, subscriptionId);
  const result = await client.fleets.createOrUpdate("rgazurefleet", "testFleet", {
    properties: {
      spotPriorityProfile: {
        capacity: 2,
        minCapacity: 1,
        evictionPolicy: "Delete",
        allocationStrategy: "PriceCapacityOptimized",
        maintain: true,
      },
      regularPriorityProfile: {
        capacity: 2,
        minCapacity: 1,
        allocationStrategy: "LowestPrice",
      },
      vmSizesProfile: [
        { name: "Standard_D2s_v3" },
        { name: "Standard_D4s_v3" },
        { name: "Standard_E2s_v3" },
      ],
      computeProfile: {
        baseVirtualMachineProfile: {
          storageProfile: {
            imageReference: {
              publisher: "canonical",
              offer: "0001-com-ubuntu-server-focal",
              sku: "20_04-lts-gen2",
              version: "latest",
            },
            osDisk: {
              caching: "ReadWrite",
              createOption: "FromImage",
              osType: "Linux",
              managedDisk: { storageAccountType: "Standard_LRS" },
            },
          },
          osProfile: {
            computerNamePrefix: "prefix",
            adminUsername: "azureuser",
            adminPassword: "TestPassword$0",
            linuxConfiguration: { disablePasswordAuthentication: false },
          },
          networkProfile: {
            networkInterfaceConfigurations: [
              {
                name: "vmNameTest",
                properties: {
                  primary: true,
                  enableAcceleratedNetworking: false,
                  ipConfigurations: [
                    {
                      name: "vmNameTest",
                      properties: {
                        subnet: {
                          id: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworks/{virtualNetworkName}/subnets/{subnetName}",
                        },
                        primary: true,
                        loadBalancerBackendAddressPools: [
                          {
                            id: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/loadBalancers/{loadBalancerName}/backendAddressPools/{backendAddressPoolName}",
                          },
                        ],
                      },
                    },
                  ],
                  enableIPForwarding: true,
                },
              },
            ],
          },
        },
        computeApiVersion: "2023-09-01",
        platformFaultDomainCount: 1,
      },
    },
    tags: { key: "fleets-test" },
    location: "eastus2euap",
  });
  console.log(result);
}
